ivy/functional/backends/tensorflow/general.py

```diff
@@ -57,6 +57,21 @@ def current_backend_str():
     return "tensorflow"
 
 
+def get_item(x: tf.Tensor, query: tf.Tensor) -> tf.Tensor:
```

Comment:
I think we discussed this before, but are you sure there is no elegant way to wrap `x.__getitem__(query)` in this case? I presume this might be more efficient than `tf.gather(x, query)` in cases where the former would work, but if the conditions to verify it works are complex then I think the approach you currently have is good.